### ISSUE TITLE
Cannot upload local charm to uncommitted model

### ISSUE THREAD

**Issue by hatched** [juju/juju-gui#3240]
When attempting to add a local charm to a new model the request fails because there is no model yet to upload to. But you also cannot create a model without a charm first added to the canvas.

As a fix we should allow the local charm to be added to the canvas then uploaded on deploy.

**Comment by frankban** [comment]
But that way wouldn't we store the whole charm contents in memory in the browser? I think maybe we should drop the user into the deployment flow with a notification that local charms can be only deployed after creating a model.

**Comment by hatched** [comment]
We may be able to extract only the config and icon files out of the zip so that we can use them prior to deploying so we wouldn't need to store anymore than those in memory.

**Comment by clagom** [comment]
Is this still relevant?

**Comment by hatched** [comment]
Yes, it'll be an issue as long as we have the 'modeling' feature.
